internal/pkg/api/server/templates/service/service.go

```diff
@@ -314,19 +319,51 @@ func (s *service) UpgradeInstanceValidateInputs(dependencies.Container, *Upgrade
 	return nil, NotImplementedError{}
 }
 
-func repositories() []model.TemplateRepository {
-	return []model.TemplateRepository{
-		{
-			Type: "git",
-			Name: repository.DefaultTemplateRepositoryName,
-			Url:  repository.DefaultTemplateRepositoryUrl,
-			Ref:  "api-demo",
-		},
-	}
+func repositories(repoPath string) ([]model.TemplateRepository, error) {
+	if repoPath == "" {
+		return []model.TemplateRepository{
+			{
+				Type: "git",
+				Name: repository.DefaultTemplateRepositoryName,
+				Url:  repository.DefaultTemplateRepositoryUrl,
+				Ref:  "api-demo",
+			},
+		}, nil
+	}
+
+	if strings.HasPrefix(repoPath, "file://") {
+		return []model.TemplateRepository{
+			{
+				Type: "dir",
+				Name: repository.DefaultTemplateRepositoryName,
+				Path: strings.TrimPrefix(repoPath, "file://"),
+			},
+		}, nil
+	}
+	if strings.HasPrefix(repoPath, "https://") {
```

Comment:
Keby si ten protokol neriesil, tak co sa stane? (mozno som to nedomyslel, tak preto sa pytam)
IMHO by mal ten `file://` fungovat rovnako ako `https://` ...
Do buducna budeme asi podporovat aj `ssh://` (+private key), tak aby sa to nemuselo prepisovat.

Reply:
Bude prázdný `Ref` což by se někde nemuselo očekávat a v https://github.com/keboola/keboola-as-code/blob/74776ae05d1fef7b79cd0231a1635b0e71010975/internal/pkg/api/server/templates/dependencies/dependencies.go#L114 by se neměl spouštět git checkout (ten `Ref` se ale používá zřejmě až v něm, tak asi jen ten jeden `if` bude třeba pořešit).

 🤔 A ještě se dívám že i když nastavím ten dir, tak se stejně checkoutuje git repo s `main` branchí. To budu muset ještě někde fixnout:
![CleanShot 2022-05-16 at 12 34 49](https://user-images.githubusercontent.com/215660/168575080-16bf4f48-f4f5-4c5d-a3f5-9a5688220a7e.jpg)

Reply:
Jo a tady tím parsováním teda ještě ověřuju, že pokud je tam git path, tak obsahuje Ref.

Reply:
No mozno by to ref nemuselo byt prazdne, a ten checkout by tiez nemusel byt problem, ... proste by bol iba iny protokol.
Pri testoch sa to celkom hodi, ze tych rozdielov bude co najmenej, nie?

Reply:
> Jo a tady tím parsováním teda ještě ověřuju, že pokud je zadaná git path, tak obsahuje Ref.
Jj, parsovat to bude potrebne tak/ci tak, proti tomu nic nemam

Reply:
No ale ten dir nemusí být git repozitář (a v testech ani není), tím pádem na něm `clone` i `checkout` failují

Reply:
Dobra poznamka, jj, je to o tom ci by sme nevynutili GIT repozitar pre API a API testy.
Bolo by tam potrebne vyriesit `.gittest -> .git` co sa riesit niekde v unit testoch.

Ok. Ak ti to pride jednoduchsie, riesit to oddelene, tak moze byt.
Len som chcel vediet, co by znamenala realne ta druha alternativa.

Reply:
Asi můžeme mergnout `Repository.Path` do `Repository.Url`, v tom problém nevidím. Ale do toho vynucování Git repozitáře na templaty bych byl pro se nepouštět. 😃 Jakákoli změna testu by byla dost oser. Přejmenovat `.gittest` na `.git`, upravit, commitnout, přejmenovat zpět `.git` na `.gittest`. Docela mi to stačilo v tom jednom unit testu. 😅
(Tím pádem bych `Repository.Type` nechal.)

Reply:
jj, pravda, ze zmena by sa robila blbo, dik za vysvetlenie :+1: